### PR TITLE
Load specific types from adapter

### PR DESCRIPTION
**Pull request by Sanan07**
Added functionality to MSTestAdapter from Test Platform, where we load only necessary types from assemblies.

### CODE REVIEW

**Review by Haplois — APPROVED** [review]

File: scripts/test.ps1 [hunk 1]
```diff
@@ -185,10 +185,13 @@ function Run-Test([string[]] $testContainers, [string[]] $netCoreTestContainers)
 
 function Get-VSTestPath
 {
-	$vsInstallPath = Locate-VsInstallPath
-	$vstestPath = Join-Path -path $vsInstallPath "Common7\IDE\CommonExtensions\Microsoft\TestWindow\vstest.console.exe"
+    $versionsFile = "$PSScriptRoot\build\TestFx.Versions.targets"
+    $TestPlatformVersion = (([XML](Get-Content $versionsFile)).Project.PropertyGroup.TestPlatformVersion).InnerText
+
+	$vsInstallPath = "$PSScriptRoot\..\packages\Microsoft.TestPlatform.$TestPlatformVersion\"
+	$vstestPath = Join-Path -path $vsInstallPath "tools\net451\Common7\IDE\Extensions\TestPlatform\vstest.console.exe"
```
Comment (Haplois) [inline scripts/test.ps1:192]:
```suggestion
	$versionsFile = "$PSScriptRoot\build\TestFx.Versions.targets"
	$testPlatformVersion = (([XML](Get-Content $versionsFile)).Project.PropertyGroup.TestPlatformVersion).InnerText
	$testPlatformPath = "$PSScriptRoot\..\packages\Microsoft.TestPlatform.$testPlatformVersion\"
	$vstestPath = Join-Path -path $testPlatformPath "tools\net451\Common7\IDE\Extensions\TestPlatform\vstest.console.exe"
```
nit: Consider fixing the formatting and variable names...

**Review by Haplois — APPROVED** [review]
LGTM, with a small nit on my own code :)

File: src/Adapter/MSTest.CoreAdapter/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -16,6 +18,7 @@
 [assembly: AssemblyTrademark("")]
 [assembly: AssemblyCulture("")]
 [assembly: NeutralResourcesLanguage("en")]
+[assembly: TypesToLoad(typeof(MSTestDiscoverer), typeof(MSTestExecutor))]
```
Comment (shyamnamboodiripad) [inline src/Adapter/MSTest.CoreAdapter/Properties/AssemblyInfo.cs:21]: I realize this feedback is coming late @Sanan07 but I didn't realize this attribute was already available :) The name of this attribute is a little confusing. `[TestExtensionTypes]` may be a better name.

Comment (Sanan07) [reply src/Adapter/MSTest.CoreAdapter/Properties/AssemblyInfo.cs:21]: My bad, I didn't add you as a reviewer. I will consider changing the attribute name. Thank you for your feedback.

Comment (shyamnamboodiripad) [reply src/Adapter/MSTest.CoreAdapter/Properties/AssemblyInfo.cs:21]: No worries - glad to know the attribute is now available! :)

**Pull request merged**